src/assets-loader/assets-loader.ts

```diff
@@ -0,0 +1,127 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { Injectable } from '@nestjs/common';
+import { AssetDescriptionType, Transaction } from '@prisma/client';
+import { AssetDescriptionsService } from '../asset-descriptions/asset-descriptions.service';
+import { AssetsService } from '../assets/assets.service';
+import { BlockDto } from '../blocks/dto/upsert-blocks.dto';
+import { BlockOperation } from '../blocks/enums/block-operation';
+import { LoggerService } from '../logger/logger.service';
+import { BasePrismaClient } from '../prisma/types/base-prisma-client';
+import { TransactionDto } from '../transactions/dto/upsert-transactions.dto';
+import { TransactionsService } from '../transactions/transactions.service';
+
+@Injectable()
+export class AssetsLoader {
+  constructor(
+    private readonly assetsService: AssetsService,
+    private readonly assetDescriptionsService: AssetDescriptionsService,
+    private readonly logger: LoggerService,
+    private readonly transactionsService: TransactionsService,
+  ) {}
+
+  async loadDescriptions(
+    block: BlockDto,
+    prisma: BasePrismaClient,
+  ): Promise<void> {
+    const main = block.type === BlockOperation.CONNECTED;
+
```

Comment:
Didn't you say the loader was going to use the PrismaService? Or is that happening in a subsequent PR:
https://github.com/iron-fish/ironfish-api/pull/1228#discussion_r1106348002

Just wondering because the replay of these might be tricky if it fails in the middle of updating a batch

Reply:
This loader is using the prisma client from the block loader so mints and burns will be processed alongside transactions in the same transaction. I should have included more detail in my comment about where the original service was being injected from.

If these get slow, I'll move it into a worker job and then add a Prisma injection.